solana-programs/anchor/audius-data/programs/audius-data/src/lib.rs

```diff
@@ -228,47 +222,13 @@ pub mod audius_data {
     }
 
     /*
-        Track related functions
+        Entity related functions
     */
-    pub fn create_track(ctx: Context<CreateTrack>, metadata: String) -> Result<()> {
-        msg!("Audius::CreateTrack");
-        if ctx.accounts.authority.key() != ctx.accounts.user.authority {
-            return Err(ErrorCode::Unauthorized.into());
-        }
-        // Set owner to user storage account
-        ctx.accounts.track.owner = ctx.accounts.user.key();
-        ctx.accounts.track.track_id = ctx.accounts.audius_admin.track_id;
-        // Increment global track ID after assignment to this track in particular
-        // Ensures each track has a unique numeric ID
-        ctx.accounts.audius_admin.track_id = ctx.accounts.audius_admin.track_id + 1;
-        msg!("AudiusTrackMetadata = {:?}", metadata);
-        Ok(())
-    }
-
-    pub fn update_track(ctx: Context<UpdateTrack>, metadata: String) -> Result<()> {
-        msg!("Audius::UpdateTrack");
-        if ctx.accounts.user.key() != ctx.accounts.track.owner {
-            return Err(ErrorCode::Unauthorized.into());
-        }
-        if ctx.accounts.authority.key() != ctx.accounts.user.authority {
-            return Err(ErrorCode::Unauthorized.into());
-        }
-        msg!("AudiusTrackMetadata = {:?}", metadata);
-        Ok(())
-    }
-
-    pub fn delete_track(ctx: Context<DeleteTrack>) -> Result<()> {
-        msg!("Audius::DeleteTrack");
-        if ctx.accounts.user.key() != ctx.accounts.track.owner {
-            return Err(ErrorCode::Unauthorized.into());
-        }
+    pub fn manage_entity(ctx: Context<ManageEntity>, _entity_type: EntityTypes, _management_action: ManagementActions, _id: String, _metadata: String) -> Result<()> {
+        // Reject if update submitted with invalid user authority
         if ctx.accounts.authority.key() != ctx.accounts.user.authority {
```

Comment:
we're going to change this to validate_user_authority next right?

Reply:
yes definitely will be an immediate change